src/resolver.ts

```diff
@@ -3347,11 +3347,10 @@ export class Resolver extends DiagnosticEmitter {
     node: NamedTypeNode,
     /** How to proceed with eventual diagnostics. */
     reportMode: ReportMode = ReportMode.REPORT
-  ): TypeNode | null { 
+  ): TypeNode | null {
     var typeArgumentNodes = node.typeArguments;
-    if (!typeArgumentNodes || typeArgumentNodes.length != 1) {
-      let numTypeArguments = 0;
-      if (typeArgumentNodes) numTypeArguments = typeArgumentNodes.length;
+    const numTypeArguments = typeArgumentNodes ? typeArgumentNodes.length : 0;
```

Comment:
That's wrong. Why you don't use my suggestion?

Reply:
I'm not a fan of assignment in a conditional.

Reply:
In this case better leave previous solution